cinema/views.py

```diff
@@ -1,58 +1,151 @@
-from rest_framework import viewsets
-
-from cinema.models import Genre, Actor, CinemaHall, Movie, MovieSession
-
+from django.db.models import F, Count
+
+from rest_framework import viewsets, status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from cinema.models import (
+    Genre,
+    Actor,
+    CinemaHall,
+    Movie,
+    MovieSession,
+    Order
+)
 from cinema.serializers import (
     GenreSerializer,
     ActorSerializer,
     CinemaHallSerializer,
     MovieSerializer,
     MovieSessionSerializer,
-    MovieSessionListSerializer,
-    MovieDetailSerializer,
     MovieSessionDetailSerializer,
     MovieListSerializer,
+    OrderListSerializer,
+    OrderSerializer,
+    MovieGetDetailSerializer,
+    MovieSessionListSerializer,
 )
 
 
 class GenreViewSet(viewsets.ModelViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
```

Comment:
The 'GenreViewSet' uses a custom 'list' method to modify the response data. Ensure that this is necessary, as it might interfere with pagination if enabled. Consider using a custom serializer if specific data formatting is needed.